featurebyte/service/base_document.py

```diff
@@ -1024,6 +1028,12 @@ async def update_document(
             return await self.get_document(document_id=document_id)
         return None
 
+    @retry(
+        retry=retry_if_exception_type(OperationFailure),
+        wait=wait_chain(
+            *[wait_random(max=RETRY_MAX_WAIT_IN_SEC) for _ in range(RETRY_MAX_ATTEMPT_NUM)]
+        ),
+    )
     async def update_documents(
         self,
         query_filter: QueryFilter,
```

Comment:
This update method involves multiple documents and it could hit mongo `WriteConflict` error. This issue is similar to feature readiness update & adding a retry should handle most of the cases.